util/ticker/ticker.go

```diff
@@ -0,0 +1,33 @@
+package ticker
+
+import (
+	"time"
+)
+
+type Ticker interface {
+	C() <-chan time.Time
+	Stop()
+	Reset(time.Duration)
+}
+
+type realTicker struct {
+	t *time.Ticker
+}
+
+func (rt *realTicker) Stop() {
+	rt.t.Stop()
+}
+
+func (rt *realTicker) Reset(d time.Duration) {
```

Comment:
None of the existing clock libraries that I found (including the one from apimachinery that's in use elsewhere) properly support mocking the `Reset` method of a ticker, so I implemented a small shim to enable me to mock it in the tests.  We can swap to the apimachinery one in the future if they add support